src/datasources/flow-ds-react/helpers.ts

```diff
@@ -556,31 +559,49 @@ const processRawSeriesData = (queryData: FlowParsedQueryRow, options: FlowQueryR
 
     const multiplier = getMultiplier(queryData, responseData.timestamps)
     const combineIngressEgress = isFunctionSet(queryData, FlowFunctionStrings.combineIngressEgress);
-
-    let processedData = responseData.columns.map((column, colIdx) => {
-        const datapoints = convertTimeStampedDataToDataFrame(timestampsInRange, colIdx, responseData, !!nanToZero);
-        return {
-            target: convertLabel(column.label, column, queryData),
-            datapoints
-        }
-    })
-
-    if (combineIngressEgress) {
-        const uniqueColumns = [...new Set(responseData.columns.map(col => col.label))]
-        processedData = uniqueColumns.map((uniqueColumn) => {
-            const datapoints = timestampsInRange.map((timestamp, tindex) => {
-                const sum = responseData.columns.filter((col) => col.label === uniqueColumn).map((_, columnIndex) => {
-                    const v = responseData.values[columnIndex][tindex];
-                    return isNaN(Number(v) && nanToZero ? 0 : v)
-                }).reduce((prv, cur) => prv + (cur ? cur : 0), 0)
-                return [sum * multiplier, timestamp]
-            })
-            const columnIndexb = responseData.columns.findIndex((d) => d.label === uniqueColumn);
+    const onlyIngress = isFunctionSet(queryData, FlowFunctionStrings.onlyIngress);
+    const onlyEgress = isFunctionSet(queryData, FlowFunctionStrings.onlyEgress);
+    const negativeIngress = isFunctionSet(queryData, FlowFunctionStrings.negativeIngress);
+    const negativeEgress = isFunctionSet(queryData, FlowFunctionStrings.negativeEgress);
+
+    let processedData = responseData.columns
+        .filter((column) => !(onlyIngress && !column.ingress || onlyEgress && column.ingress))
+        .map((column, colIdx) => {
+            const sign = negativeIngress && column.ingress || negativeEgress && !column.ingress ? -1 : 1;
+            const datapoints = convertTimeStampedDataToDataFrame(timestampsInRange, colIdx, responseData, !!nanToZero, multiplier, sign);
             return {
-                target: convertLabel(uniqueColumn as string, responseData.columns[columnIndexb], queryData),
+                target: convertLabel(column.label, column, queryData),
                 datapoints
             }
         })
+
+    if (combineIngressEgress) {
+        const uniqueColumns = [...new Set(responseData.columns.map(col => col.label))]
+        const columnsWithIndex = responseData.columns.map((column, colIdx) => { return { column, colIdx } });
+
+        processedData = uniqueColumns
+            .map((uniqueColumn) => {
+
+                const datapoints = timestampsInRange
```

Comment:
Can we break this out into its own function? We've hit more levels of indentation here than I'm comfortable with.

Reply:
Also, can we use more descriptive variable names (sum of what, what column index, prv/cur context), move the inline comments into a jsdoc comment explaining what the whole block is doing. I want to move away from the very long functions in the existing Helm code.